crates/refunder/src/lib.rs

```diff
@@ -29,9 +40,53 @@ pub async fn main(args: arguments::Arguments) {
     loop {
         tracing::info!("Staring a new refunding loop");
         match refunder.try_to_refund_all_eligble_orders().await {
-            Ok(_) => (),
-            Err(err) => tracing::error!("Error while refunding ethflow orders: {:?}", err),
+            Ok(_) => {
+                track_refunding_loop_result("success");
+                *liveness.last_successful_loop.write().expect(
+                    "Lock is only written to at this point and the program should just crash on a panic",
```

Comment:
I would expect this to never panic because the lock should never be poisoned (i.e. there is no opportunity of a panic unwinding while a lock is held).

Reply:
I would do a plain unwrap without explanation. This type of unwrap is pretty ocmmon.

Reply:
To me this wasn't obvious since `shared::metrics::serve_metrics` might in theory write into the `RwLock` and panic there. I considered passing to `Liveness` a "wrapped" `RwLock` that can only be read to be certain that it can't be poisoned, but I didn't find a way to do it nicely and I thought a manual implementation was overkill.

Anyway, I removed this line and replaced it with an `unwrap()`.